validphys2/src/validphys/commondataparser.py

```diff
@@ -70,22 +100,72 @@ def parse_commondata(commondatafile, systypefile, setname):
         nkin=3,
         nsys=nsys,
         commondata_table=commondatatable,
-        systype_table=systypetable
+        systype_table=systypetable,
     )
 
+
 def parse_systypes(systypefile):
-    """Parses a systype file and returns a pandas dataframe.
-    """
+    """Parses a systype file and returns a pandas dataframe."""
     systypeheader = ["sys_index", "type", "name"]
     try:
         systypetable = pd.read_csv(
             systypefile, sep=r"\s+", names=systypeheader, skiprows=1, header=None
         )
-        systypetable.dropna(axis='columns', inplace=True)
+        systypetable.dropna(axis="columns", inplace=True)
     # Some datasets e.g. CMSWCHARMRAT have no systematics
     except pd.errors.EmptyDataError:
         systypetable = pd.DataFrame(columns=systypeheader)
 
     systypetable.set_index("sys_index", inplace=True)
 
     return systypetable
+
+
+@dataclasses.dataclass
```

Comment:
These should be frozen by default. We may want to be caching based on this a.t some point

Reply:
```suggestion
@dataclasses.dataclass(frozen=True)
```